modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -523,4 +642,38 @@ private String _interpolateURL(
 	@Reference
 	private ReactRenderer _reactRenderer;
 
+	private static class ObjectEntryComparator
+		implements Comparator<ObjectEntry> {
+
+		public ObjectEntryComparator(List<Long> ids) {
+			_ids = ids;
+		}
+
+		@Override
+		public int compare(ObjectEntry objectEntry1, ObjectEntry objectEntry2) {
+			long id1 = objectEntry1.getId();
+			long id2 = objectEntry2.getId();
+
+			int index1 = _ids.indexOf(id1);
+			int index2 = _ids.indexOf(id2);
+
+			if ((index1 == -1) && (index2 == -1)) {
+				return Long.compare(id1, id2);
+			}
+
+			if (index1 == -1) {
+				return 1;
+			}
+
+			if (index2 == -1) {
+				return -1;
+			}
+
+			return Long.compare(index1, index2);
+		}
+
+		private final List<Long> _ids;
+
+	}
+
```

Comment:
[Not a blocker for PR] Yesterday I stumbled upon the issue this code fixes, and [fixed it in another way](https://github.com/markocikos/liferay-portal/commit/4becdf94591a6350c2c14d390aa52f3eca6c2dc4), arguably simpler. The solution in this PR is still more than fine. My solution might be worth considering if Brian doesn't like this, or some issues pop up.